### PR TITLE
Assortment of patches from my arm-3.0 branch

### PR DESCRIPTION
**Pull request by gohai**

### CODE REVIEW

**Comment by gohai** [comment]
Rebased to current master so that it applies cleanly

**Comment by gohai** [comment]
@benfry How is this looking? (I hope <target if="" does what I think it does)

**Comment by gohai** [comment]
@benfry Something like this? I hope `<target if=""` does what I think it does..

File: build/build.xml [hunk 1]
```diff
@@ -24,6 +24,21 @@
...
+  <!-- detetect Raspberry Pi and friends -->
+  <condition property="linux-arm32" value="linux-arm32">
+    <and>
+      <equals arg1="${platform}" arg2="linux" />
+      <equals arg1="${os.arch}" arg2="arm" />
+      <equals arg1="${sun.arch.data.model}" arg2="32" />
+    </and>
+  </condition>
+
+  <target name="check-linux-arm32" if="linux-arm32">
```
Comment (benfry) [inline build/build.xml]: I think you still need an `antcall` or another target that depends on this one so that it's called properly. (Or am I missing that elsewhere in this PR?)

File: build/build.xml [hunk 2]
```diff
@@ -219,10 +216,22 @@
 
   <target name="jre-download" depends="jre-check, downloader-setup"
           unless="jre.tgz.downloaded">
+    <!-- we normally download a JRE, except on arm -->
+    <property name="jre.downloadJdk" value="false" />
```
Comment (benfry) [inline build/build.xml]: Every variable in build.xml uses dots, not camel case. Let's keep it that way.

File: core/build.xml [hunk 3]
```diff
@@ -39,13 +44,16 @@
 	   encoding="UTF-8"
 	   includeAntRuntime="false"
 	   debug="true"
-	   srcdir="src" destdir="bin"
```
Comment (benfry) [inline core/build.xml]: Why is this removed?

File: build/build.xml (hunk 1 above)
Comment (gohai) [inline build/build.xml]: Made linux-build depend on his target below

File: core/build.xml [hunk 4]
```diff
@@ -39,13 +44,16 @@
...
+	   destdir="bin"
 	   classpath="apple.jar; 
                       library/jogl-all.jar; 
                       library/gluegen-rt.jar"
 	   nowarn="true">
       <!-- kinda gross, but if not using the JDT, this just ignored anyway -->
       <compilerclasspath path="${jdt.jar}; ../java/mode/jdtCompilerAdapter.jar" />
+      <src path="src"/>
+      <include name="processing/**"/>
+      <exclude name="processing/javafx/**" if="fx.missing"/>
```
Comment (benfry) [inline core/build.xml]: Let's call it `fx.unavailable`. And `if` can be used here? (Didn't know that...)

File: build/build.xml (hunk 2 above)
Comment (gohai) [inline build/build.xml]: Will change!

File: core/build.xml (hunk 4 above)
Comment (gohai) [inline core/build.xml]: Will rename. Didn't know that either - but it works even on my prehistoric ant on the Pi, which doesn't support the regular `<if>` elements.
